libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -64,14 +63,9 @@ abstract class DefaultApolloExtension(
   }
 
   internal fun registerDownstreamProject(serviceName: String, projectPath: String) {
-    val existingService = services.firstOrNull {
-      it.name == serviceName
-    }
-    if (existingService != null) {
-      existingService.isADependencyOf(project.rootProject.project(projectPath))
-    } else {
-      pendingDownstreamDependencies.compute(serviceName) { _, oldValue ->
-        oldValue.orEmpty() + projectPath
+    project.configurations.configureEach {
+      if (it.name == ModelNames.configuration(serviceName, ApolloDirection.Downstream, ApolloUsage.Ir, ConfigurationKind.DependencyScope)) {
+        it.dependencies.add(project.dependencies.project(mapOf("path" to projectPath)))
       }
```

Comment:
This is the actual fix. If `existingService` was not null, it's too late to call `isADependencyOf()`.